app/pages/project/components/ProjectNavbar/ProjectNavbar.jsx

```diff
@@ -1,76 +1,57 @@
 import _ from 'lodash';
-import getRenderedSize from 'react-rendered-size';
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import withSizes from 'react-sizes';
+import sizeMe from 'react-sizeme'
 import ProjectNavbarNarrow from './components/ProjectNavbarNarrow';
-import ProjectNavbarWide from './components/ProjectNavbarWide';
+import ProjectNavbarWide, { SizeAwareProjectNavbarWide } from './components/ProjectNavbarWide';
 
 function haveNavLinksChanged(oldProps, newProps) {
-  const oldNavLinks = _.map(oldProps.navLinks, 'label');
-  const newNavLinks = _.map(newProps.navLinks, 'label');
-  return _.difference(newNavLinks, oldNavLinks).length > 0 || oldNavLinks.length !== newNavLinks.length;
-}
-
-function hasTitleChanged(oldProps, newProps) {
-  return oldProps.projectTitle !== newProps.projectTitle;
+  const oldLinks = _.map(oldProps.navLinks, 'label');
+  const newLinks = _.map(newProps.navLinks, 'label');
+  return _.difference(oldLinks, newLinks).length > 0 ||
+    oldLinks.length !== newLinks.length;
 }
 
 export class ProjectNavbar extends Component {
   constructor(props) {
     super(props);
+    this.setBreakpoint = this.setBreakpoint.bind(this);
     this.state = {
-      breakpoint: 0
+      useWide: false
     };
   }
 
-  componentDidMount() {
-    this.setBreakpoint();
-  }
-
   componentDidUpdate(prevProps) {
-    if (haveNavLinksChanged(prevProps, this.props) ||
-      hasTitleChanged(prevProps, this.props)) {
+    if (haveNavLinksChanged(prevProps, this.props)) {
       this.setBreakpoint();
     }
   }
 
-  setBreakpoint() {
-    const sizes = getRenderedSize(
-      <ProjectNavbarWide
-        {...this.props}
-        style={{
-          position: 'absolute',
-          visibility: 'hidden'
-        }}
-      />
-    );
-
-    this.setState({
-      breakpoint: sizes.width
-    });
+  setBreakpoint(size) {
+    // `size` is undefined when the component is first mounted, as there hasn't
+    // been time for the callback to fire.
+    if (size) {
+      const useWide = size.width < document.body.clientWidth;
+      this.setState({ useWide });
+    }
   }
 
   render() {
-    const isWindowWide = this.props.width > this.state.breakpoint;
-    if (isWindowWide) {
-      return <ProjectNavbarWide {...this.props} />;
-    }
-
-    return <ProjectNavbarNarrow {...this.props} />;
+    const NavBarComponent = (this.state.useWide) ? ProjectNavbarWide : ProjectNavbarNarrow;
+
+    return (
+      <div>
+        <NavBarComponent {...this.props} />
+        <SizeAwareProjectNavbarWide
+          {...this.props}
+          onSize={this.setBreakpoint}
+          style={{
+            visibility: 'hidden',
+            position: 'absolute'
```

Comment:
If `useWide` is true, I think you should remove these styles and show this nav bar, rather than rendering it twice in the DOM.

Reply:
If I remove these styles, the visible navbar goes into narrow mode, even if there's enough room to render the wide navbar. I think this happens because of a couple of things:
 - `position: absolute` roughly halves the rendered size of the project navbar (see screenshots).
 - `document.body.clientWidth` is smaller than the viewport, so will always be smaller than an element that complete fills the window.

Here's what the size-aware component looks like, if we remove `visibility: hidden`.
Without `position: absolute`:
![Screenshot showing both project navbars rendered one above the other.](https://user-images.githubusercontent.com/59547/39429619-50d5c2c6-4c83-11e8-9180-f35a98d28270.png)

With `position: absolute`.
![Screenshot showing how absolute positiion reduces the width of the navigation.](https://user-images.githubusercontent.com/59547/39429773-df8905a0-4c83-11e8-9eed-1e19c7d93ea8.png)

Reply:
Using a separate rendering of the navbar makes sense to me for the purposes of dynamically setting the breakpoint as new props are received. Absolute positioning it gives us the width of the children. I'm not sure why Space Warps was collapsing into the narrow version here. My guess is because the default state is `useWide: false` and the callback to get the size measurement hadn't updated it yet. I've updated this branch to have a loading state rather than show a default of the narrow navbar which seems to have fixed this. I was seeing a flash of it before, but I think the component needs to wait for the setBreakpoint callback to fire first.

Reply:
When the narrow version was rendered, `document.body.clientWidth` was 1280 and the wide navbar width was also 1280, so the dropdown is being shown if the wide navbar exactly fills the screen. This is the missing `=` in the comparison that I picked up in a separate comment.

If I set `width: 99%` on the second navbar, the first example renders the wide navbar too.

Reply:
It's not obvious from the code that we're measuring the minimum width of the wide navbar and comparing that with the document inner width. I thought that the code was measuring the total space that the wide navbar would take up if it was rendered as the project navigation. Better variable naming might make this clearer to someone coming to this code for the first time.